frameworks/kafka/docs/security.md

```diff
@@ -0,0 +1,242 @@
+
+*Note*: It is possible to enable Kerberos after initial installation but the service may be unavailable during the transition. Additionally, your Kafka clients will need to be reconfigured.
+
+
+### SSL Auth
+
+SSL Auth requires that all clients be they brokers, producers, or consumers present a valid TLS Certificate from which their identity can be derived. DC/OS Apache Kafka uses the `CN` of the TLS certificate as the principal for a given client. For example, the certificate `CN=bob@example.com,OU=,O=Example,L=London,ST=London,C=GB` will be considered as the principal `bob@example.com`.
+
+#### Prerequisites
+- Competion of the section [Transport Encryption](#transport-encryption) above
+
+#### Install the Service
+
+Install the DC/OS Apache Kafka service with the following options alongside your own:
+```
+{
+    "service": {
+        "service_account": "<service-account>",
+        "service_account_secret": "<secret path>"
+        "security": {
+            "transport_encryption": {
+                "enabled": true
+            },
+            "ssl_authentication": {
+                "enabled": true
+            }
+        }
+    }
+}
+```
+
+*Note*: It is possible to enable SSL Auth after initial installation, but the service may be unavailable during the transition. Additionally, your Kafka clients will need to be reconfigured.
+
+#### Authenticating a Client
+
+To authenticate a client to DC/OS Apache Kafka, you will need to configure it to use a TLS certificate signed by the DC/OS CA. After generating a certificate signing request, you can issue it to the DC/OS CA by calling the API `<dcos-cluster>/ca/api/v2/sign`. Using `cURL` the request would look like:
+```
+curl -X POST \
+-H "Authorization: token=$(dcos config show core.dcos_acs_token)" \
+<dcos-cluster>/ca/api/v2/sign \
+-d '{"certificate_request": "<json-encoded-value-of-request.csr>"}'
+```
+
+The response will contain a signed public certificate. Full details on the DC/OS CA API can be found [here](https://docs.mesosphere.com/latest/security/ent/tls-ssl/ca-api/).
+
+## Authorization
+
+The DC/OS Apache Kafka Service supports Kafka's ACL based authorization (authz) system. To use Kafka's authz, either SSL Auth or Kerberos must be enabled as detailed above.
+
+### Enable Authorization
+
+#### Prerequisites
+- Completion of either [Recovery Plan](#ssl-auth) or [Kerberos](#kerberos-authn) above.
+
+#### Install the Service
+
+Install the DC/OS Apache Kafka service with the following options alongside your own (remember, either SSL Auth or Kerberos _must_ be enabled):
+```
+{
+    "service": {
+        "security": {
+            "authorization": {
+                "enabled": true,
+                "super_users": "<list of super users>",
+                "allow_everyone_if_no_acl_found": <true|false default false>
```

Comment:
Should we add a discussion of this setting?

Reply:
I'm not sure. What do you think makes sense to say?

Reply:
"The `allow_everyone_if_no_acl_found` setting allows all authenticated users to access topics with no authorization configured. The alternative is that only super users are able to access such topics. This can be used when transitioning from a cluster with no authorization enabled to a secure cluster (see [Kafka Documentation](http://kafka.apache.org/documentation/#security_authz)"? (at least link to the docs)?